src/app/admin/results/level-results-chart/level-results-chart.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@angular/core';
+import { ApiService } from 'src/app/shared/services/api.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LevelResultsChartService {
+
+  constructor(private apiService: ApiService) { }
+  /** returns data for questions with true answers */
+  getCorrectAnswers(data) {
+    const questionsArr = JSON.parse(data).filter(element =>
+      element.true === 1
+    ).map(element => element.question_id);
+    return this.apiService.getByEntityManager('Question', questionsArr);
+  }
+  /** returns TestDetail for test with provided id */
+  getTestDetail(id) {
+    return this.apiService.getEntityByAction('TestDetail', 'getTestDetailsByTest', id);
+  }
+  /** returns object with counted true answers for each level */
+  getTrueAnswersbyLevel(testDetails, trueAnswers): Array<object> {
+    const arr = [];
+    let i = 0;
+    testDetails.forEach(element => {
```

Comment:
Can you use map here? You could use index as map parameter instead of i.

Reply:
fixed